### PR TITLE
[controller] Allow restricting certain DatadogAgent features 

### PR DESCRIPTION
**Pull request by xornivore**
### What does this PR do?

Introduces additional CLI options to allow restricting:
- DatadogAgent features, such as running system-probe container
- Container image registries
- Using specific host storage path for temp storage in logs agent.

### Motivation

Better support for managed Kubernetes environments.

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: controllers/datadogagent/utils.go [hunk 1]
```diff
@@ -973,12 +973,17 @@ func getVolumesForAgent(dda *datadoghqv1alpha1.DatadogAgent) []corev1.Volume {
 	}
 
 	if datadoghqv1alpha1.BoolValue(dda.Spec.Agent.Log.Enabled) {
+
+		var tempStoragePath = *dda.Spec.Agent.Log.TempStoragePath
+		if options.AgentHostStoragePath != "" {
+			tempStoragePath = filepath.Join(options.AgentHostStoragePath, "logs")
+		}
 		volumes = append(volumes, []corev1.Volume{
```
Comment (clamoriniere) [inline controllers/datadogagent/utils.go:981]: I'm wondering if we can hide this behind a function if we go for a variadic options approach

```golang
package options

func GetHostStoragePath(options ...ReconcilerOptions, defaultPath string) string
```

Comment (xornivore) [reply controllers/datadogagent/utils.go:981]: In any case we should add a helper to abstract it a bit.

File: controllers/datadogagent/utils.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ func getAuthTokenSecretName(dda *datadoghqv1alpha1.DatadogAgent) string {
 }
 
 // newAgentPodTemplate generates a PodTemplate from a DatadogAgent spec
-func newAgentPodTemplate(logger logr.Logger, dda *datadoghqv1alpha1.DatadogAgent, selector *metav1.LabelSelector) (*corev1.PodTemplateSpec, error) {
+func newAgentPodTemplate(logger logr.Logger, dda *datadoghqv1alpha1.DatadogAgent, selector *metav1.LabelSelector, options *ReconcilerOptions) (*corev1.PodTemplateSpec, error) {
```
[the remark this answers is not in the document]
Comment (xornivore) [reply controllers/datadogagent/utils.go:55]: I like the idea but perhaps we should do it in a separate set of changes.

**Comment by clamoriniere** [comment]
The code as too much changed since this PR was open, lets close it. 

feel free to reopen it if you think the opposite

**Pull request closed without merging**